### DISCUSSION TITLE
Angular: Required Typescript Version not respected

### DISCUSSION THREAD

**Discussion by flodaniel** [renovatebot/renovate#10030] [Request Help] [answered]
Hello, 

I am totally new to renovate. I created a self-hosted setup for our small company on Bitbucket Cloud. 
The bot created the `all-minor` PR, but wants to update typescript from `~4.0.2 `to `~4.2.0`, which breaks the build, as Angular requires `<4.2.0`. How can I make sure that renovate does not update to Versions that are out of the required range, given by other packages? 

We are using drone.io for automatic builds, which has the following error on the branch: 
`ERROR in The Angular Compiler requires TypeScript >=4.0.0 and <4.2.0 but 4.2.4 was found instead.`

Any help is greatly appreciated :) 

This is my config.json: 
```
{
  "platform": "bitbucket",
  "password": "*****",
  "username": "*****",
  "autodiscover": true,
  "autodiscoverFilter": "******",
  "gitAuthor": "Renovate Bot <*****>",
  "baseBranches": ["develop"],
}
```

This is my default.json which is a shared preset: 

```
{
  "docker-compose": {
    "enabled": false
  },
  "dockerfile": {
    "enabled": false
  },
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "matchUpdateTypes": ["minor", "patch"],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch"
    },
    {
      "groupName": "eslint",
      "matchPackagePatterns": ["^eslint"]
    },
    {
      "groupName": "angular-eslint",
      "matchPackagePatterns": ["^@angular-eslint/"]
    },
    {
      "groupName": "typescript-eslint",
      "matchPackagePatterns": ["^@typescript-eslint/"]
    },
    {
      "groupName": "fortawesome",
      "matchPackagePatterns": ["^@fortawesome/"]
    },
    {
      "groupName": "angular",
      "matchPackagePatterns": ["^@angular/"]
    }
  ]
}

```

**Comment by viceice** [comment, marked as answer]
I'm using a custom preset for angular projects

https://github.com/visualon/renovate-config/blob/main/ng.json

**Reply by viceice** [reply]
```json
   {
      "description": "Require typescript minor approval for angular",
      "matchPackageNames": ["typescript"],
      "matchUpdateTypes": ["minor"],
      "dependencyDashboardApproval": true 
   },
```

**Reply by flodaniel** [reply]
Interesting. This is more a workaround? Does the bot not look into the package-lock.json and evalute required versions?

**Reply by viceice** [reply]
nope, we do not check transitive deps. It would be very complicated.

**Reply by flodaniel** [reply]
@viceice I implemented the same config as you, however there seems to be a slight problem. It now creates a seperate PR for typescript, and typescript update is removed from the PR description of my `all-minor` PR. But it seems like it is not correctly updating/resetting the package.json, as in there the typescript version has been changed before I used your config:

Typescript not listed: 
![image](https://user-images.githubusercontent.com/6661557/118617818-463f6e80-b7c3-11eb-83fa-b7506db95077.png)


But still changed in package.json

![image](https://user-images.githubusercontent.com/6661557/118617867-535c5d80-b7c3-11eb-9930-3d169bec3fa7.png)

**Reply by rarkins** [reply]
Try ticking the rebase/retry checkbox

**Reply by flodaniel** [reply]
I just noticed, that the docker image of the renovate bot is shutdown after the initial run. How do I keep it alive, so i can trigger a rebase by the bot, without restarting the docker image? Platform is bitbucket cloud.

**Reply by rarkins** [reply]
That's not supported
